### PR TITLE
feat(smartcontract, activator): remove multiple tunnel restriction (#2725)

### PR DESCRIPTION
**Pull request by bgm-malbeclabs**
## Summary of Changes
This PR adds the ability from the CLI and the smart contract to add simultaneous tunnels: a single unicast and a single mulitcast tunnel (subscriber or publisher only) on the same or different devices.

Closes https://github.com/malbeclabs/doublezero/issues/2735

## Testing Verification
* E2E tests were extended to include creating multiple tunnels for a single user
* The coexistence tests that were skipped are now enabled and pass
* This also adds tests to verify that `pubkey` is always the last deserialized field in both go and rust
* Tests to validate the default ip and edge cases
* Claude helped extend the tests

### CODE REVIEW

File: client/doublezero/src/command/connect.rs [hunk 1]
```diff
@@ -462,59 +460,148 @@ impl ProvisioningCliCommand {
...
+                        group_pk: *mcast_group_pk,
+                        client_ip: *client_ip,
+                        publisher,
+                        subscriber,
+                    })?;
+                }
+
+                user_pk
+            }
+            // Both IBRL and Multicast users exist - add subscription to existing Multicast user
```
[the remark this answers is not in the document]
Comment (bgm-malbeclabs) [reply client/doublezero/src/command/connect.rs]: yea great catch

**Comment by bgm-malbeclabs** [comment]
> > ```
> >   const AIRDROP_USER_RENT_LAMPORTS_BYTES: usize = 240 * 3; // 240 bytes per User account
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To account for the 4 bytes for the newly added `tunnel_endpoint`.
> 
> Maybe we can add a test for this so it doesn't get forgotten is User grows?

added a test - good call out. Also added tests to validate go and rust deserialization given the requirement for pubkey to be the last field

File: e2e/ibrl_multicast_coexistence_test.go [hunk 2]
```diff
@@ -87,6 +78,331 @@ func TestE2E_IBRL_AllocatedAddr_Multicast_Publisher_Coexistence(t *testing.T) {
...
+		"deviceIP", actualMcastDevice.CYOANetworkIP,
+		"ibrlDeviceCode", ibrlDevice.Spec.Code,
+		"ibrlDeviceID", ibrlDevice.ID,
+		"ibrlDeviceIP", ibrlDevice.CYOANetworkIP)
+	waitForAgentConfigWithClient(t, log, dn, actualMcastDevice, client)
+	log.Info("--> Agent config pushed to multicast device")
+
+	// Give time for agent to apply the configuration
+	log.Info("==> Waiting for agent to apply configuration")
+	time.Sleep(10 * time.Second)
```
[the remark this answers is not in the document]
Comment (bgm-malbeclabs) [reply e2e/ibrl_multicast_coexistence_test.go]: yea probably not

File: activator/src/process/user.rs [hunk 3]
```diff
@@ -238,13 +287,68 @@ pub fn process_user_event(
             )
             .unwrap();
 
+            // Determine tunnel endpoint: if the client demanded a specific one, validate it;
```
Comment (snormore) [inline activator/src/process/user.rs]: Want to put this into a method if it's the same logic/code as above

Comment (bgm-malbeclabs) [reply activator/src/process/user.rs]: updated

**Review by snormore — APPROVED** [review]
